tests/data/sieve-vacation-on-no-end-date.txt

```diff
@@ -11,8 +11,8 @@ if address "From" "marketing@company.org" {
 }
 
 ### Nextcloud Mail: Vacation Responder ### DON'T EDIT ###
-# DATA: {"version":1,"enabled":true,"start":"2022-09-02","subject":"On vacation","message":"I'm on vacation."}
```

Comment:
could you also add/keep tests for parsing the previous markup?

we could also leverage the `version` spec if that is helpful. if you configure the account on two instances with mail v3.6 and v3.4 and set the timezone aware spec with v3.6, the v3.4 instance will have trouble parsing the new format, I guess.

Reply:
Well, I forgot to push some changes. That explains why tests ran locally.

---

That is a good point. I think we don't need a version bump as the new format (ISO 8601) is supported by the JS's `Date` constructor and PHP's `DateTime` constructor.

I'll also add some test that parse the old format without time zones to make sure that it works.

Reply:
I'll quickly test what happens when loading the script in two different versions.

Reply:
It just works :tm: 



**Note:** The last day is off by one on stable3.4 but that is expected. As I said earlier, the new logic actually includes the last day (which was broken before) and thus the old Mail version picks up the date as `2023-12-13T00:00:00` which is correct (as we want to respond until `2023-12-12` is actually over).

I fixed the picker in this PR to subtract a day (*visually*) to not confuse the user. The last day is actually inclusive (and always has been).

Saving the exact same form in stable3.4 and loading it again on this PR still shows the same data as shown below. I tested that. So we even have full compatibility and don't need to break anything with the version.

**Left:** this PR, **Right:** stable3.4

![grafik](https://github.com/nextcloud/mail/assets/1479486/2444150c-7938-4650-a716-887c5abb5120)